ooniprobe/View/OONIRun/OoniRunViewController.m

```diff
@@ -190,16 +190,35 @@ -(void)validateAndAddURLs{
     }
 }
 
-/*
- This regex tests whether the URL starts with http:// or https://,
- then checks for at least 1 character, then checks for a dot, and then again checks for at least 1 character.
- No spaces allowed.
+/**
+ * Checks if the input string is a valid URL with a maximum length of 2083 characters and enforces that the scheme is only "http" or "https".
+ *
+ * @param urlString The input string to check if it is a valid URL.
+ * @return YES if the input string is a valid URL; otherwise, NO.
  */
-- (BOOL)validateLink:(NSString *)link
-{
-    NSString *regex = @"(?i)(http|https)(:\\/\\/)([^ .]+)(\\.)([^ \n]+)";
-    NSPredicate *predicate = [NSPredicate predicateWithFormat:@"SELF MATCHES %@", regex];
-    return [predicate evaluateWithObject:link];
+- (BOOL)validateLink:(NSString *)urlString {
+    // Check if the length of the input string is greater than 2083 characters; if so, the URL is not valid
+    if (urlString.length > 2083) {
```

Comment:
I would suggest removing this constraint on the maximum URL length. AFAIK this is a limitation that only used to exist in older internet explorer versions, but it's not really something part of the specification for URLs.